scripts/gt_concordance.py

```diff
@@ -0,0 +1,551 @@
+                    'a1': a1,
+                    'a2': a2 }
+    return gt_data
+
+def check_concordance(sample, vcf_data, gt_data, config):
+    project = sample.split('_')[0]
+    matches = []
+    mismatches = []
+    lost = []
+    for chromosome_position in vcf_data.keys():
+        chromosome, position = chromosome_position.split()
+        vcf_a1 = vcf_data[chromosome_position]['a1']
+        vcf_a2 = vcf_data[chromosome_position]['a2']
+        if chromosome_position not in gt_data:
+            log.warning('POSITION {} NOT FOUND IN GT DATA!!!'.format(chromosome_position))
+            continue
+
+        gt_a1 = gt_data[chromosome_position]['a1']
+        gt_a2 = gt_data[chromosome_position]['a2']
+        concordance = set([gt_a1, gt_a2]) == set([vcf_a1, vcf_a2])
+        if concordance:
+            matches.append([chromosome, position, vcf_a1, vcf_a2, gt_a1, gt_a2])
+        else:
+            if gt_a1 != '0' and gt_a2 != '0':
+                mismatches.append([chromosome, position, vcf_a1, vcf_a2, gt_a1, gt_a2])
+            else:
+                lost.append([chromosome, position, vcf_a1, vcf_a2, gt_a1, gt_a2])
+
+    # calculating concordance and round to 2 decimals
+    percent_matches=round((float(len(matches))/float(len(vcf_data) - len(lost))*100), 2)
+
+    # sort by chromosome and position
+    matches = sorted(matches, key=lambda x:(int(x[0]) if x[0] != 'X' else x[0], int(x[1])))
+    mismatches = sorted(mismatches, key=lambda x:(int(x[0]) if x[0] != 'X' else x[0], int(x[1])))
+    lost = sorted(lost, key=lambda x:(int(x[0]) if x[0] != 'X' else x[0], int(x[1])))
+
+    # recording results
+    result = '{}\n'.format(sample)
+    result += 'Chrom Pos A1_seq A2_seq A1_maf A2_maf\n'
+    result += 'Mismatches: {}\n'.format(len(mismatches))
+    result += '\n'.join([' '.join(mismatch) for mismatch in mismatches])
+    result += '\nLost: {}:\n'.format(len(lost))
+    result += '\n'.join([' '.join(lost_snp) for lost_snp in lost])
+    result += '\n'
+    result += '\nLost snps: {}\n'.format(len(lost))
+    result += 'Total number of matches: {} / {} / {}\n'.format(len(matches), len(vcf_data)-len(lost), len(vcf_data))
+    result += 'Percent matches {}%\n'.format(percent_matches)
+
+
+    # path should exist (if we came to this point), but checking anyway
+    output_path = os.path.join(config.get('ANALYSIS_PATH'), project, 'piper_ngi/03_genotype_concordance')
+    if os.path.exists(output_path):
+        # create .conc file
+        with open(os.path.join(output_path, '{}.conc'.format(sample)), 'w+') as conc_file:
+            conc_file.write(result)
+
+    if len(matches) + len(mismatches) + len(lost) != len(vcf_data):
+        log.warning('CHECK RESULTS!! Numbers are incoherent. Total number of positions: {}, matches: {}, mismatches: {}, lost: {}'.format(len(vcf_data), len(matches), len(mismatches), len(lost)))
+
+    if len(lost) >= 30:
+        log.warning('Too few positions in VCF file!! Failed to caclulate concordance')
```

Comment:
maybe you should do the same thing you do when you do not have enough position, you foce to 0 the concordance